pkg/vdr/web/resolver.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package web
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+)
+
+// Read resolves a did:web did.
+func (v *VDR) Read(didID string, opts ...vdr.ResolveOpts) (*did.Doc, error) {
+	// apply resolve opts
+	docOpts := &vdr.ResolveDIDOpts{
+		HTTPClient: &http.Client{},
+	}
+
+	for _, opt := range opts {
+		opt(docOpts)
+	}
+
+	url, err := parseDIDWeb(didID)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:web did --> could not parse did:web did --> %w", err)
+	}
+
+	resp, err := docOpts.HTTPClient.Get(url)
```

Comment:
If the http client is configured properly, it will validate the certificate against the configured root authorities. This is good.

Is this also ensuring this part of the `did:web` spec (important):

> The method specific identifier MUST match the common name used in the SSL/TLS certificate, and it MUST NOT include IP addresses or port numbers.

So for instance if you resolve `did:web:acme.org` and the server's certificate's common name is "Faber College" - does it fail or not? This might be a followup PR - can you confirm?

Reply:
After some digging through the http and tls packages I don't think we explicitly check for that at the moment. Would this just require a comparison between the two (common cert name and method specific identifier)?

Reply:
@amodkala

> Would this just require a comparison between the two (common cert name and method specific identifier)?

Roughly, yes.

Reply:
@amodkala are we updating this PR any time soon? Or would we rather add this verification in a followup?